cmake/ccache.cmake

```diff
@@ -0,0 +1,29 @@
+#[[
+Copyright 2023 Antony Peacock
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+]]
+
+include_guard(GLOBAL)
+
+if (${MORPHEUS_BUILD_WITH_CCACHE})
+    find_program(CCACHE_BIN ccache REQUIRED)
+    if(CCACHE_BIN)
+        message(STATUS "Morpheus: CCache found: ${CCACHE_BIN}. Enabling ccache as compiler cache tool.")
+        set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_BIN}")
```

Comment:
It seems this setting is intended to only be used by CTest: https://cmake.org/cmake/help/latest/prop_gbl/RULE_LAUNCH_COMPILE.html

If we use `[<LANG>_COMPILER_LAUNCHER]` then we should loop through all supported languages setting it for each and every language selected (we should allow this to be selected by input values to the function).

Reply:
My bad, I took the suggestion from this [presentation](https://www.meetingcpp.com/mcpp/slides/2021/building_and_packaging_modern_cpp231.pdf). It was probably referring to old settings of Conan 1.X.
Thanks, I will update the logic accordingly.

Reply:
No worries, I only know this as this is the approach advocated by the one of the CMake authors in his book Professional CMake: https://crascit.com/professional-cmake/